### DISCUSSION TITLE
[feature_request]  Add all time total commits object to GraphQL API

### DISCUSSION THREAD

**Discussion by rickstaa** [community/community#35675] [Apps, API and Webhooks]
There is currently no easy way to fetch the all-time commits of a specific user using the GraphQL API. The current `user/contributionsCollection/totalCommitContributions` object of the GraphQL API will return the commit contributions per year (see [contributionsCollection](https://docs.github.com/en/graphql/reference/objects#contributionscollection)). Providing users to retrieve the all-time commit contributions without having to loop through all years would improve the GraphQL API.

# Current behaviour

**Current graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    contributionsCollection {
      totalCommitContributions
      restrictedContributionsCount
    }
  }
}
```

**Current GraphQL variables**

```graphql
{"login": "rickstaa"}
```

This will return the commits per year.

# New behaviour

Making the `totalCommitContributions` and `restrictedContributionsCount` objects available under the user level would improve the behaviour.

**New graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    totalCommitContributions
    restrictedContributionsCount
    contributionsCollection {
      totalCommitContributions
      restrictedContributionsCount
    }
  }
}
```

**New GraphQL variables**

```graphql
{"login": "rickstaa"}
```

#### Github REST API

I can fetch the total commits of a user by using the commit search API to get this in a single request: https://developer.github.com/v3/search/#search-commits it has a `total_count` field.

e.g.
```
curl -H "Accept: application/vnd.github.cloak-preview" https://api.github.com/search/commits\?q\=author:adamyi
```

but it has a rate limit of 30 queries per minute and sometimes fails.

### Using github pagination

I can also use [GraphQL pagination](https://graphql.org/learn/pagination/) to loop through all [ContributionsCollection.contributionYears](https://docs.github.com/en/graphql/reference/objects#contributionscollection). This however will take a long time and is in-accurate (see https://github.com/orgs/community/discussions/35675#discussioncomment-3844912).

**Comment by Rongronggg9** [comment]
**I second the proposal as there is no other reliable way to fetch one's overall commits on GitHub.**

> I can fetch the total commits of a user by using the commit search API to get this in a single request: [developer.github.com/v3/search/#search-commits](https://developer.github.com/v3/search/#search-commits) it has a total_count field.

Nope, it is inaccurate. Most kernel developers will probably not be in love with it as I've described in https://github.com/anuraghazra/github-readme-stats/issues/564#issuecomment-1083765940.

```sh
curl -H "Accept: application/vnd.github.cloak-preview" https://api.github.com/search/commits\?q\=author:torvalds
```

```
"total_count": 212410082,
```

**Comment by Olf0** [comment]
**I concur with both prior posts.**
Unfortunately no one at GitHub has replied yet, at least acknowledging the issue this feature request addresses.

I also second the assessment in the second post, that the function(s) offered by GitHub's REST API do(es) *not* provide an alternative.
Furthermore, GitHub's GraphQL API shall provide a complete ("exhaustive") set of functions for the functionality it offers, IMO.

**Comment by mathbunnyru** [comment]
It would be nice to have this feature implemented. Currently, if you have commits in popular repos and they are copied (without fork), your commits will be counted multiple times, and the number of commits doesn't make sense anymore.
I would appreciate if someone from GitHub could take a look here.
